File: exercise/144/iteration.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> result;
        stack<TreeNode*> nodes;
        nodes.push(root);
        while (!nodes.empty()) {
            TreeNode *now = nodes.top();
            nodes.pop();
            if (!now) continue;
            result.push_back(now->val);
            nodes.push(now->right, now->left);
        }
        returen result;
    }
};

File: exercise/Binary-Tree/144/iteration.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def preorderTraversal(self, root: TreeNode) -> List[int]:
        result = []
        stack = [root]
        while len(stack) > 0:
            now = stack.pop()
            if now is not None:
                result.append(now.val)
                stack.append(now.right)
                stack.append(now.left)
        return result
